src/main/java/com/heeverse/ticket_order/service/reader/StreamAggregationReader.java

```diff
@@ -0,0 +1,47 @@
+package com.heeverse.ticket_order.service.reader;
+
+import com.heeverse.ticket.domain.entity.Ticket;
+import com.heeverse.ticket.domain.mapper.TicketMapper;
+import com.heeverse.ticket_order.domain.dto.persistence.AggregateSelectMapperDto;
+import com.heeverse.ticket_order.domain.mapper.TicketOrderAggregationMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.groupingBy;
+
+/**
+ * @author gutenlee
+ * @since 2023/10/17
+ */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class StreamAggregationReader implements AggregationReader {
+
+    private final TicketMapper ticketMapper;
+    private final TicketOrderAggregationMapper aggregationMapper;
+
+
+    public List<AggregateSelectMapperDto.Response> getResultGroupByGrade(AggregateSelectMapperDto.Request request) {
+
+        List<Ticket> tickets = ticketMapper.findTickets(request.concertSeq());
+
+        Map<String, List<Ticket>> collected
+                = tickets.stream()
+                .collect(groupingBy(Ticket::getGradeName));
+
+
+        return collected.entrySet().parallelStream()
+                .map(entry -> {
+                    List<Long> seqList = entry.getValue().stream().map(Ticket::getSeq).collect(Collectors.toList());
+                    List<AggregateSelectMapperDto.Response> responses = aggregationMapper.selectByTicketSeqList(seqList);
+                    log.info("{} : order Try {}", entry.getKey(), responses.size());
+                    return new AggregateSelectMapperDto.Response(request.concertSeq(), entry.getKey(), seqList.size(), responses.size());
+                }).collect(Collectors.toList());
```

Comment:
stream 체이닝 망식 도중 코드블럭으로 로직이 나올 경우 가독성이 많이 떨어지는 상황이 발생하게 됩니다.
이럴경우 이런 로직을 메서드로 분리하여 stream 체이닝을 최대한 간결하게 유지해주는 것이 좋습니다.
```suggestion
                .map(~~method).collect(Collectors.toList());
```